### PR TITLE
Allow reserved words as column names for CREATE TABLE statements

### PR DESCRIPTION
**Pull request by zachmu**
Needs tests

### CODE REVIEW

File: go/vt/sqlparser/parse_test.go [hunk 1]
```diff
@@ -2961,14 +2961,74 @@ func TestCreateTable(t *testing.T) {
...
+				t.Errorf("input: %s, err: %v", tcase.input, err)
+				return
+			}
+			if got, want := String(tree.(*DDL)), tcase.output; got != want {
+				t.Errorf("Parse(%s):\n%s, want\n%s", tcase.input, got, want)
+			}
+		})
+	}
+
+	nonsupportedKeywords := []string{
```
Comment (zachmu) [inline go/vt/sqlparser/parse_test.go:2976]: What does non supported mean? Do they work in MySQL?

The ones that work in MySQL should have a skipped test.

**Review by zachmu — COMMENTED** [review]
Just one comment otherwise LG

File: go/vt/sqlparser/parse_test.go (hunk 1 above)
Comment (max-hoffman) [reply go/vt/sqlparser/parse_test.go:2976]: i tried to add quick fixes for as many as i could, i haven't tested all of them with mysql

Comment (max-hoffman) [reply go/vt/sqlparser/parse_test.go:2976]: i'll run them through mysql. what's the optimization trade-off for getting the long tail of mysql-supported statements?

Comment (max-hoffman) [reply go/vt/sqlparser/parse_test.go:2976]: I wrote a script to check the remaining keywords -- mysql supports all of them. Added skipped tests to be more explicit.

File: go/vt/sqlparser/parse_test.go [hunk 2]
```diff
@@ -2961,14 +2961,74 @@ func TestCreateTable(t *testing.T) {
...
+	for _, x := range nonsupportedKeywords {
+		nonsupported[x] = true
+	}
+
+	for key := range keywords {
+		input := fmt.Sprintf("create table t (%s bigint)", key)
+		output := fmt.Sprintf("create table t (\n\t`%s` bigint\n)", key)
+		t.Run(input, func(t *testing.T) {
+			if _, ok := nonsupported[key]; ok {
+				t.Skipf("Dolt does not currently support keyword as column name: %s", key)
```
Comment (zachmu) [inline go/vt/sqlparser/parse_test.go:3021]: Not "dolt"

**Review by zachmu — COMMENTED** [review]
LG, just one small comment (I can't approve because it's my PR)

**Pull request merged**